packages/@aws-cdk/aws-codebuild/lib/project.ts

```diff
@@ -385,14 +418,14 @@ export interface CommonProjectProps {
    *
    * @default - No asset build script.
    */
-  readonly buildScriptAsset?: assets.Asset;
+  readonly buildScript?: assets.ZipDirectoryAsset;
 
   /**
    * The script in the asset to run.
    *
    * @default build.sh
    */
-  readonly buildScriptAssetEntrypoint?: string;
+  readonly buildScriptEntrypoint?: string;
```

Comment:
Not needed. Let's just support an asset source type and let users specify buildSpec.

We can create a subclass of `Project` like `ScriptProject` if we want a higher level abstraction to just run a script from an asset.

Reply:
See above - I don't understand what you mean by this comment.